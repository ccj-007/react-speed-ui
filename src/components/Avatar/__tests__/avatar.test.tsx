import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Icon, Avatar } from '../../index'
import { solid } from "@fortawesome/fontawesome-svg-core/import.macro";
describe('test avatar component', () => {
  it('use string & icon & img should render', () => {
    const { asFragment } = render(
      <>
        <Avatar text='User'></Avatar>
        <Avatar icon={<Icon icon={solid("user")} size="1x" color="#fff"></Icon>}></Avatar>
        <Avatar style={{ width: '30px', height: '30px' }} src='https://gimg2.baidu.com/image_search/src=http%3A%2F%2Fc-ssl.duitang.com%2Fuploads%2Fitem%2F201708%2F07%2F20170807114113_4efNW.thumb.1000_0.jpeg&refer=http%3A%2F%2Fc-ssl.duitang.com&app=2002&size=f9999,10000&q=a80&n=0&g=0n&fmt=auto?sec=1665131372&t=27759fd5e80cffad11405f01579fb6a1'></Avatar>
      </>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('different size should render', () => {
    const { asFragment } = render(
      <>
        <Avatar style={{ background: 'green' }} size={40} text='A'></Avatar>
        <Avatar style={{ background: 'orange' }} size={60} icon={<Icon icon={solid("user")} size="2x" color="#fff"></Icon>} ></Avatar>
      </>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('different shape should render', () => {
    const { asFragment } = render(
      <>
        <Avatar shape="circle" style={{ background: 'green' }} size={40} text='A'></Avatar>
        <Avatar shape="circle" style={{ background: 'orange' }} size={60} icon={<Icon icon={solid("user")} size="2x" color="#fff"></Icon>} ></Avatar>
      </>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
