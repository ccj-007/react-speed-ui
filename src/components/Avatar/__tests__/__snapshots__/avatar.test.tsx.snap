// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test avatar component different shape should render 1`] = `
<DocumentFragment>
  <div
    class="speed-avatar"
    style="width: 40px; height: 40px; border-radius: 50%; background: green;"
  >
    <div
      style="background: green;"
    >
      A
    </div>
  </div>
  <div
    class="speed-avatar"
    style="width: 60px; height: 60px; border-radius: 50%; background: orange;"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-user fa-2x speed-icon"
      color="#fff"
      data-icon="user"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0S96 57.3 96 128s57.3 128 128 128zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
        fill="currentColor"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`test avatar component different size should render 1`] = `
<DocumentFragment>
  <div
    class="speed-avatar"
    style="width: 40px; height: 40px; border-radius: none; background: green;"
  >
    <div
      style="background: green;"
    >
      A
    </div>
  </div>
  <div
    class="speed-avatar"
    style="width: 60px; height: 60px; border-radius: none; background: orange;"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-user fa-2x speed-icon"
      color="#fff"
      data-icon="user"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0S96 57.3 96 128s57.3 128 128 128zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
        fill="currentColor"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`test avatar component use string & icon & img should render 1`] = `
<DocumentFragment>
  <div
    class="speed-avatar"
    style="width: 30px; height: 30px; border-radius: none;"
  >
    <div>
      U
    </div>
  </div>
  <div
    class="speed-avatar"
    style="width: 30px; height: 30px; border-radius: none;"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-user fa-1x speed-icon"
      color="#fff"
      data-icon="user"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0S96 57.3 96 128s57.3 128 128 128zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="speed-avatar"
    style="width: 30px; height: 30px; border-radius: none;"
  >
    <img
      alt=""
      src="https://gimg2.baidu.com/image_search/src=http%3A%2F%2Fc-ssl.duitang.com%2Fuploads%2Fitem%2F201708%2F07%2F20170807114113_4efNW.thumb.1000_0.jpeg&refer=http%3A%2F%2Fc-ssl.duitang.com&app=2002&size=f9999,10000&q=a80&n=0&g=0n&fmt=auto?sec=1665131372&t=27759fd5e80cffad11405f01579fb6a1"
      style="width: 30px; height: 30px;"
    />
  </div>
</DocumentFragment>
`;
